### PR TITLE
Use QT for OpenCV GUI Backend When It's Enabled

### PR DESCRIPTION
**Pull request by mattdawkins**

### CODE REVIEW

File: CMake/fletch-tarballs.cmake [hunk 1]
```diff
@@ -315,6 +279,42 @@ if (fletch_ENABLE_Qt OR fletch_ENABLE_VTK OR fletch_ENABLE_qtExtensions OR
...
+    set(OpenCV_md5 "886b0c511209b2f3129649928135967c")
+  else()
+    message(ERROR " OpenCV Version \"${OpenCV_version}\" Not Supported")
+  endif()
+else()
+  # Remove Contrib repo option when OpenCV is not enabled
+  unset(fletch_ENABLE_OpenCV_contrib CACHE)
+endif()
+list(APPEND fletch_external_sources OpenCV)
+
```
Comment (dstoup) [inline CMake/fletch-tarballs.cmake]: I'm not seeing the change between these 2 big blocks of code. What is the diff?

**Comment by kfieldho** [comment]
@dstoup @Cookt2 Please take a look.  I've rebased this and tweaked it to work with current master.  I've also added the ability to NOT use the fletch Qt even if its available just to keep it consistent with other similar situations.  BTW, the large move of code in fletch-tarballs is to move OpenCV after Qt which, because of the way fletch-tarballs is handled is required.

**Comment by Cookt2** [comment]
LGTM.. except for the mysterious boost error from the windows build. I'm not convinced that error is related to this branch. @dstoup Have you seen that boost error before?

**Comment by Cookt2** [comment]
Jenkins test this please

**Comment by Cookt2** [comment]
running the windows build again..

**Comment by dstoup** [comment]
Jenkins test this please

**Comment by dstoup** [comment]
Jenkins test this please

**Comment by dstoup** [comment]
Jenkins test this please

**Pull request merged**